source/common/json/json_streamer.h

```diff
@@ -21,7 +21,7 @@ namespace Json {
  */
 class Streamer {
 public:
-  using Value = absl::variant<absl::string_view, double, uint64_t, int64_t>;
+  using Value = absl::variant<absl::string_view, double, uint64_t, int64_t, bool>;
```

Comment:
Would you mind plopping some static asserts in the constructor somewhere to ensure the type ordering at the various indices is what we think it is? Probably right above where you've got the switch statement that relies on the index() being 4 for bools.

It should look approximately like:
```
static_assert(std::is_same<decltype(absl::get<4>(value)), bool>::value, "value at index 4 must be bool");
```

I know I'm being paranoid, but just in case someone down the line ends up changing the order without modifying the switch statement, we can catch it at compile-time.

Reply:
+1 -- good idea @tonya11en

Reply:
Very good call. Thank you.